frontend/src/metabase-lib/v1/expressions/helper-text-strings.ts

```diff
@@ -432,6 +432,44 @@ const HELPER_TEXT_STRINGS: HelpTextConfig[] = [
       },
     ],
   },
+  {
+    name: "month-name",
+    structure: "monthName",
+    description: () =>
+      t`Returns the localized short name ("Apr") for the given month number (4)`,
+    args: [
+      {
+        name: t`monthNumber`,
+        description: t`Column or expression giving the number of a month in the year, 1 to 12.`,
+        example: formatIdentifier(t`Birthday Month`),
+      },
+    ],
+  },
+  {
+    name: "quarter-name",
+    structure: "quarterName",
+    description: () => t`Returns a string like "Q1", given the quarter number`,
+    args: [
+      {
+        name: t`quarterNumber`,
+        description: t`Column or expression giving the number of a quarter of the year, 1 to 4.`,
+        example: formatIdentifier(t`Fiscal Quarter`),
+      },
+    ],
+  },
+  {
+    name: "day-name",
+    structure: "dayName",
+    description: () =>
+      t`Returns the localized name of a day of the week, given the day's number`,
```

Comment:
Can/should we make it clear if localization includes handling various first days of the week (e.g., Sunday vs Monday)?

Reply:
I thought about that too, and was intending to check it on this PR and see if more work is needed.

It's tricky because in the common case this is a Column Extraction from a datetime column straight to day name: `[:day-name {} [:get-day-of-week {} ...]]`. In that case, it doesn't actually matter which day is `1`, just that they're consistent between the two functions.

But since you could use either function separately, we can't just rely on that. There's `metabase.public-settings/start-of-week`, and we should be respecting that here.[1]

That might mean the values need to be adjusted and can't just be translated directly with `java.time.*`, but that's what `metabase.shared.util.time` is for. I'll keep digging into this and try to get everything consistent with the setting.


[1] There's a bunch of machinery in the QP to handle the variation here. `driver.common/start-of-week-offset` is powered by `driver/db-start-of-week`, that's what drives the adjustment. So we already expect this to be inconsistent, and juggle the expressions in SQL to fix it, so that what comes out of the `:get-day-of-week` expression lines up with `metabase.public-settings/start-of-week`.

Reply:
Check out the fancy new test for this!